components/match2/commons/match_group_input.lua

```diff
@@ -250,4 +251,29 @@ function MatchGroupInput.fetchStandaloneMatch(matchId)
 	end)
 end
 
+--[[
+Coerces the opponent specification part of an opponent record. If the opponent
+spec is not present, then the record will be treated as an input args table,
+from which the spec will be parsed using Opponent.readOpponentArgs.
+]]
+function MatchGroupInput.coerceOpponentRecord(record)
+	local opponent = Opponent.readOpponentArgs(record)
+	if opponent.type == 'team' then
```

Comment:
This should use the types as defined in #663

Reply:
It's probably better to run a global search and replace on all files in the repo, instead of changing it piecemeal manually. It's out of scope for this pr.

Reply:
This is new code though, you can use it here.

Reply:
Ok

Reply:
Thought I already made this change - turns out I didn't